storage/gcp/bucket/variables.tf

```
@@ -0,0 +1,156 @@
+  description = "The bucket's Versioning configuration."
+  type        = object({
+    enabled = bool
+  }) 
+  default     = null
+}
+
+variable "website" {
+  description = "Configuration if the bucket acts as a website. Structure is documented below."
+  type        = object({
+    main_page_suffix = optional(string)
+    not_found_page   = optional(string)
+  })
+  default     = null
+}
+
+variable "cors" {
+  description = "The bucket's Cross-Origin Resource Sharing (CORS) configuration."
+  type        = object({
+    origin          = optional(list(string))
+    method          = optional(list(string))
+    response_header = optional(list(string))
+    max_age_seconds = optional(number)
+  })
+  default = null
+}
+
+variable "default_event_based_hold" {
+  description = "Whether or not to automatically apply an eventBasedHold to new objects added to the bucket."
+  type        = bool 
+  default     = false 
+}
+
+variable "retention_policy" {
+  description = "Configuration of the bucket's data retention policy for how long objects in the bucket should be retained."
+  type        = object({
+    is_locked        = optional(bool)
+    retention_period = number
+  })
+  default = null
+}
+
+variable "labels" {
+  description = "A map of key/value label pairs to assign to the bucket."
+  type        = map(string)
+  default     = {}
+}
+
+variable "logging" {
+  description = "The bucket's Access & Storage Logs configuration."
+  type        = object({
+    log_bucket        = string
+    log_object_prefix = optional(string)
+  })
+  default = null
+}
+
+variable "requester_pays" {
+  description = "Enables Requester Pays on a storage bucket."
+  type        = bool
+  default     = false
+}
+
+variable "encryption" {
+  description = "The bucket's encryption configuration."
+  type        = object({
+    default_kms_key_name = string
+  })
+  default = null
+}
+
+variable "uniform_bucket_level_access" {
+  description = "Enables Uniform bucket-level access access to a bucket"
+  type        = bool
+  default     = false
+}
+
+variable "public_access_prevention" {
+  description = "Prevents public access to a bucket. Acceptable values are 'inherited' or 'enforced'"
+  type        = string
+  default     = "inherited"
+}
+
+variable "custom_placement_config" {
+  description = "The bucket's custom location configuration, which specifies the individual regions that comprise a dual-region bucket. If the bucket is designated a single or multi-region, the parameters are empty."
+  type        = object({
+    data_locations = list(string)
+  })
+  default = null
+}
+
+############### SECTION - Bucket acl
+
+variable "role_entity" {
+  description = "List of role/entity pairs for acls bucket"
+  type        = list(string)
+  default     = []
+}
+
+############### SECTION - Bucket policy
+
+variable "policy_data" {
+  description = "Policy data to bind to the bucket"
+  type        = map(list(string))
+  default     = null
+}
```

In GCP, we talk about roles instead of permissions.
```suggestion
############### SECTION - Bucket permissions

variable "roles" {
  description = "Roles to bind to the bucket"
  type        = map(set(string))
  default     = null
}
```

If you want a term that is provider neutral, you can use "permission".